examples/sources/Picasso_ExplicitMomentumUpdate.hpp

```diff
@@ -0,0 +1,460 @@
+    }
+};
+
+//---------------------------------------------------------------------------//
+// Update nodal momentum n+1 with stress, gravity
+//---------------------------------------------------------------------------//
+struct UpdateGridVelocity
+{
+    // Explicit time step size.
+    double dt;
+
+    template <class LocalMeshType, class GatherDependencies,
+              class ScatterDependencies, class LocalDependencies>
+    KOKKOS_INLINE_FUNCTION void
+    operator()( const LocalMeshType&, const GatherDependencies& gather_deps,
+                const ScatterDependencies&, const LocalDependencies&,
+                const int i, const int j, const int k ) const
+    {
+        // Get the local dependencies.
+        auto m_i =
+            gather_deps.get( Picasso::FieldLocation::Node(), Example::Mass() );
+        auto u_i = gather_deps.get( Picasso::FieldLocation::Node(),
+                                    Example::Velocity() );
+        auto delta_u_s_i = gather_deps.get( Picasso::FieldLocation::Node(),
+                                            Example::DeltaUStress() );
+
+        auto delta_u_g_i = gather_deps.get( Picasso::FieldLocation::Node(),
+                                            Example::DeltaUGravity() );
+
+        // Compute velocity.
+        Picasso::Vec3<double> zeros = { 0.0, 0.0, 0.0 };
+        u_i( i, j, k ) +=
+            ( m_i( i, j, k ) > 0.0 )
+                ? dt * ( delta_u_s_i( i, j, k ) + delta_u_g_i( i, j, k ) ) /
+                      m_i( i, j, k )
+                : zeros;
+
+        printf( "%d %d %d %8.5e %8.5e %8.5e %8.5e\n", i, j, k, m_i( i, j, k ),
+                delta_u_s_i( i, j, k, 2 ), delta_u_g_i( i, j, k, 2 ),
+                u_i( i, j, k, 2 ) );
+    }
+};
+
+//---------------------------------------------------------------------------//
+// Compute boundary condition on grid.
+//---------------------------------------------------------------------------//
+template <class BCType, class FieldType>
+struct ApplyBoundaryCondition
+{
+    BCType bc;
+
+    template <class LocalMeshType, class GatherDependencies,
+              class ScatterDependencies, class LocalDependencies>
+    KOKKOS_INLINE_FUNCTION void
+    operator()( const LocalMeshType&, const GatherDependencies&,
+                const ScatterDependencies&, const LocalDependencies& local_deps,
+                const int i, const int j, const int k ) const
+    {
+        // Get the local dependencies.
+        auto f_i =
+            local_deps.get( Picasso::FieldLocation::Node(), FieldType() );
+
+        bc.apply( f_i, i, j, k );
+    }
+};
+
+//---------------------------------------------------------------------------//
+// Explicit momentum time stepper
+//---------------------------------------------------------------------------//
+
+template <class Mesh, class InterpolationType, class ParticleVelocity,
+          class StressProperties>
+class ExplicitMomentumIntegrator
```

Comment:
Remove

Reply:
Do we want to put everything struct into the main `dam_break.cpp`?

Reply:
Yes, but combined as much as we can (maybe just one for grid updates and one for BC)